x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -214,6 +288,37 @@ export function DatatableComponent(props: DatatableRenderProps) {
     return <EmptyPlaceholder icon={LensIconChartDatatable} />;
   }
 
+  const visibleColumns = props.args.columns.columnIds.filter((field) => !!field);
+  const columnsReverseLookup = firstTable.columns.reduce<
+    Record<string, { name: string; index: number; meta?: DatatableColumnMeta }>
+  >((memo, { id, name, meta }, i) => {
+    memo[id] = { name, index: i, meta };
+    return memo;
+  }, {});
+
+  const { sortBy, sortDirection } = props.args.columns;
+
+  const rows = firstTable?.rows || [];
+  let sortedRows = rows;
+  const isReadOnlySorted = props.renderMode !== 'edit';
+
+  if (sortBy && sortDirection !== 'none') {
+    // Sort on raw values for these types, while use the formatted value for the rest
+    const sortingCriteria = ['number', 'date'].includes(
+      columnsReverseLookup[sortBy]?.meta?.type || ''
+    )
+      ? sortBy
+      : (row: Record<string, unknown>) => formatters[sortBy]?.convert(row[sortBy]);
```

Comment:
This is not taking into account ip sorting, right? If Visualize datatable isn't doing it right either at the moment, I'm fine with resolving this later, but we should create an issue for it.

Reply:
Update a first version of IPv4 sorting specific logic